src/test/java/io/orangebeard/listener/TestSuiteTreeTest.java

```diff
@@ -0,0 +1,67 @@
+package io.orangebeard.listener;
+
+import org.assertj.core.api.Assertions;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import java.util.Optional;
+import java.util.UUID;
+
+@ExtendWith(MockitoExtension.class)
+public class TestSuiteTreeTest {
+
+    @Test
+    public void when_a_child_is_requested_and_it_is_present_the_child_is_returned() {
+        TestSuiteTree root = new TestSuiteTree("", null);
+        TestSuiteTree expected = root.addChild("child", null);
+        Optional<TestSuiteTree> child2 = root.getChildByName("child");
+        Assertions.assertThat(child2.isPresent()).isTrue();
+        TestSuiteTree actual = child2.get();
+        Assertions.assertThat(actual).isEqualTo(expected);
```

Comment:
I prefer inlining the expected and the actual, so I can read on one line what's being asserted. 

Furthermore, ti would be nice to static import the Assertions class. helps with the readability as well. 

AND finally: split up the setting up of testdata, doing the test and asserting the results by empty lines.